### PR TITLE
feat: Introduce My Collection view options

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Feature**

This PR resolves [CX-1989]

### Description

This PR introduces My Collection view options (list view and grid view).

| List View | Grid View |
| --- | --- |
| Simulator Screen Shot - iPhone 12 mini - 2022-02-14 at 13 28 58 | Simulator Screen Shot - iPhone 12 mini - 2022-02-14 at 13 29 01 |

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [x] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- feat: Introduce My Collection view options (behind feature flag) - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-1989]: https://artsyproduct.atlassian.net/browse/CX-1989?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ec630fcac2f0b2cd651f050b8c8c0a263900f9b4
</p>

File: src/lib/Scenes/MyCollection/MyCollection.tsx [hunk 1]
```diff
@@ -82,6 +89,10 @@ const MyCollection: React.FC<{
     })
   }
 
+  useEffect(() => {
+    relay.loadMore(100)
+  }, [])
```
Comment (olerichter00) [inline src/lib/Scenes/MyCollection/MyCollection.tsx:94]: Currently, we load up to 100 artworks when the user opens the My Collection screen in order to be able to filter the artworks locally. With this change, we'll load 10 artworks initially and delay fetching "all" artworks in order to render earlier.

**Review by MrSltun — APPROVED** [review]
LGTM! Great work!! 🔥 ✈️

File: src/lib/Scenes/Search/UserPreferencesModel.tsx [hunk 2]
```diff
@@ -15,23 +16,27 @@ const DEFAULT_CURRENCY =
     (currencies as unknown as string[]).includes(userCurrency)
   ) as Currency) ?? "USD"
 const DEFAULT_METRIC = ""
-
+const DEFAULT_VIEW_OPTION = "grid"
 // please update this when adding new user preferences
 export interface UserPreferences {
   pricePaidCurrency: Currency
   metric: Metric
+  artworkViewOption: ViewOption
```
Comment (lordkiz) [inline src/lib/Scenes/Search/UserPreferencesModel.tsx:24]: Unless you intend to apply this to all artwork lists, then you might need to be specific with the name. How about myCollectionArtworkViewOption?

File: src/lib/Scenes/MyCollection/MyCollection.tsx (hunk 1 above)
Comment (lordkiz) [reply src/lib/Scenes/MyCollection/MyCollection.tsx:94]: However, I wonder if it is worth making double trips for the same data we can fetch all at once?

File: src/lib/Scenes/MyCollection/MyCollection.tests.tsx [hunk 3]
```diff
@@ -124,6 +129,34 @@ describe("MyCollection", () => {
...
+
+      act(() => fireEvent.press(tree.root.findByType(GridViewIcon)))
+
+      expect(MyCollectionArtworkList).toBeDefined()
+
+      act(() => fireEvent.press(tree.root.findByType(ListViewIcon)))
+
+      expect(MyCollectionArtworkList).toBeDefined()
+    })
+  })
```
Comment (lordkiz) [inline src/lib/Scenes/MyCollection/MyCollection.tests.tsx:158]: Nice test 👏🏾👏🏾👏🏾👏🏾👏🏾

**Review by lordkiz — COMMENTED** [review]
Amazing work 👏🏾 👏🏾 👏🏾 👏🏾👏🏾. Left a few small comments.

File: src/lib/Scenes/MyCollection/MyCollection.tsx (hunk 1 above)
Comment (olerichter00) [reply src/lib/Scenes/MyCollection/MyCollection.tsx:94]: This is a connection so we'll fetch artwork 0-9 then render and then fetch artwork 10-100.

**Pull request merged**